test/test.js

```diff
@@ -24,6 +61,18 @@ describe('test', function () {
     assert.isFalse(test(1));
     assert.isFalse(test(null));
     assert.isFalse(test(undefined));
+    // special characters in parameter names are not allowed
```

Comment:
I don't really remember why I excluded `gen-delims` but allowed `sub-delims`:
```abnf
gen-delims     =  ":" / "/" / "?" / "#" / "[" / "]" / "@"
```

Reply:
I remember now:

`gen-delims`: It represents a collection of reserved characters that serve as general delimiters in URIs. These characters have special significance in the structure of a URI.

They must be percent encoded not to change the semantic of URI structure.

Reply:
I think this is now clear. It comes from the section https://datatracker.ietf.org/doc/html/rfc3986#section-3.3.

```abnf
pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
```

`pchar` doesn't contain gen-delims and this library is aligned with it.